pkg/ledger/ledger.go

```diff
@@ -88,12 +87,28 @@ func (l *Ledger) Commit(ctx context.Context, txsData []core.TransactionData) (*C
 		}
 	}
 
-	l.monitor.CommittedTransactions(ctx, l.store.Name(), result)
-	return result, nil
+	if addOps != nil && addOps.SetAccountMeta != nil {
+		for addr, m := range addOps.SetAccountMeta {
+			if err := l.store.UpdateAccountMetadata(ctx,
+				addr, m, time.Now().Round(time.Second).UTC()); err != nil {
```

Comment:
You could use the same date for transactions and metadata update. After all, it is fundamentally made at the same time.